HW2/P3/mandelbrot.pyx

```diff
@@ -32,15 +35,66 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
     assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
 
     with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
+        for i in prange(in_coords.shape[0], schedule='static', chunksize=1, num_threads=4):
+        #for i in range(in_coords.shape[0]):
+            for j in range(0,in_coords.shape[1],8):
+                #c = in_coords[i, j]
+                #z = 0
+                c_real = AVX.make_float8( in_coords[i,j+7].real, in_coords[i,j+6].real,
+                                               in_coords[i,j+5].real, in_coords[i,j+4].real,
+                                               in_coords[i,j+3].real, in_coords[i,j+2].real,
+                                               in_coords[i,j+1].real, in_coords[i,j+0].real  )
+                c_imag = AVX.make_float8( in_coords[i,j+7].imag, in_coords[i,j+6].imag,
+                                               in_coords[i,j+5].imag, in_coords[i,j+4].imag,
+                                               in_coords[i,j+3].imag, in_coords[i,j+2].imag,
+                                               in_coords[i,j+1].imag, in_coords[i,j+0].imag  )
+                z_real = AVX.float_to_float8(0.0)
+                z_imag = AVX.float_to_float8(0.0)
+                cp2    = AVX.float_to_float8(2.0)
+                final_count = AVX.float_to_float8(0.0)
+                
                 for iter in range(max_iterations):
-                    if magnitude_squared(z) > 4:
+                #    if magnitude_squared(z) > 4:
+                #        break
+                #    z = z * z + c
+                #out_counts[i, j] = iter
+                    temp1 = AVX.mul(z_real,z_real)           
+                    temp2 = AVX.mul(z_imag,z_imag)           
+                    magnitude  = AVX.add( temp1,temp2 )      
+                    magnitude  = AVX.sqrt( magnitude )
+
+
+                    mask = AVX.less_than(magnitude,cp2)
```

Comment:
Directly comparing the squared terms to 4 may be a good alternative that avoids calling sqrt.